app/script/links/LinkPreviewProtoBuilder.coffee

```diff
@@ -21,8 +21,18 @@ z.links ?= {}
 
 z.links.LinkPreviewProtoBuilder = do ->
 
+  has_valid_attributes = (data) ->
+    has_valid_description = true if data.description
+    has_valid_title = true if data.title
+    has_valid_type = true if data.type in ['article', 'object', 'website']
+    has_valid_type = true if not data.type?
+    has_valid_url = true if data.url
+    return true if has_valid_description and has_valid_title and has_valid_type and has_valid_url
+    return false
```

Comment:
actually this is enough since there are `and`s :)

```coffeescript
return has_valid_description and has_valid_title and has_valid_type and has_valid_url
```